server/room.ts

```diff
@@ -941,4 +943,38 @@ export class Room {
     delete this.tsMap[socket.id];
     // delete nameMap[socket.id];
   };
+
+  private kickUser = async (
+    socket: Socket,
+    data: {
+      uid: string;
+      token: string;
+      userToBeKicked: string;
+    }
+  ) => {
+    const decoded = await validateUserToken(
+      data?.uid as string,
+      data?.token as string
+    );
+    if (!decoded) {
+      socket.emit('errorMessage', 'Failed to authenticate user');
+      return;
+    }
+    const isOwner = await this.validateOwner(decoded.uid);
+    if (!isOwner) {
+      socket.emit('errorMessage', 'Not current room owner');
+      return;
+    }
+    const userToBeKickedSocket = this.io.of(this.roomId).connected[
```

Comment:
is this an alias of https://socket.io/docs/v3/server-api/#namespace-sockets? I didn't see "connected" in the docs

Reply:
oh looks like we are using 2.0. . . maybe we should think about updating